### PR TITLE
tree-wide: voidify sd_event_source_set_enabled() calls

### PR DESCRIPTION
**Pull request by bluca**
CID#1465793
CID#1465794
CID#1465795

### CODE REVIEW

**Review by yuwata — COMMENTED** [review]
Hmm, rather than voidify the calls, I think it is better to handle errors...

**Comment by bluca** [comment]
> Hmm, rather than voidify the calls, I think it is better to handle errors...

I don't know enough about these two use cases - they are ignoring other errors, so I assumed it was intentional to skip them

File: src/journal-remote/journal-remote.c [hunk 1]
```diff
@@ -243,13 +243,13 @@ int journal_remote_add_source(RemoteServer *s, int fd, char* name, bool own_name
                 r = sd_event_add_defer(s->events, &source->buffer_event,
                                        dispatch_raw_source_until_block, source);
                 if (r == 0)
-                        sd_event_source_set_enabled(source->buffer_event, SD_EVENT_OFF);
+                        (void) sd_event_source_set_enabled(source->buffer_event, SD_EVENT_OFF);
```
Comment (poettering) [inline src/journal-remote/journal-remote.c]: i'd catch the error here and handle it below.

File: src/journal-remote/journal-remote.c [hunk 2]
```diff
@@ -243,13 +243,13 @@ int journal_remote_add_source(RemoteServer *s, int fd, char* name, bool own_name
...
         } else if (r == -EPERM) {
                 log_debug("Falling back to sd_event_add_defer for fd:%d (%s)", fd, name);
                 r = sd_event_add_defer(s->events, &source->event,
                                        dispatch_blocking_source_event, source);
                 if (r == 0)
-                        sd_event_source_set_enabled(source->event, SD_EVENT_ON);
+                        (void) sd_event_source_set_enabled(source->event, SD_EVENT_ON);
```
Comment (poettering) [inline src/journal-remote/journal-remote.c]: ditto

File: src/libsystemd/sd-event/sd-event.c [hunk 3]
```diff
@@ -3664,7 +3664,7 @@ static int source_dispatch(sd_event_source *s) {
         if (s->n_ref == 0)
                 source_free(s);
         else if (r < 0)
-                sd_event_source_set_enabled(s, SD_EVENT_OFF);
+                (void) sd_event_source_set_enabled(s, SD_EVENT_OFF);
```
Comment (poettering) [inline src/libsystemd/sd-event/sd-event.c]: turning an event source should always work. maybe enclose this in assert_se( >= 0);

File: src/libsystemd/sd-event/sd-event.c [hunk 4]
```diff
@@ -3705,7 +3705,7 @@ static int event_prepare(sd_event *e) {
                 if (s->n_ref == 0)
                         source_free(s);
                 else if (r < 0)
-                        sd_event_source_set_enabled(s, SD_EVENT_OFF);
+                        (void) sd_event_source_set_enabled(s, SD_EVENT_OFF);
```
Comment (poettering) [inline src/libsystemd/sd-event/sd-event.c]: ditto

**Review by poettering — CHANGES_REQUESTED** [review]

File: src/journal-remote/journal-remote.c (hunk 1 above)
Comment (bluca) [reply src/journal-remote/journal-remote.c]: ok, done

**Pull request merged**